### ISSUE TITLE
Reported errors differ depending on platform

### ISSUE THREAD

**Issue by katsika** [PyCQA/pylint#3489]
### Steps to reproduce
1. `git clone git@github.com:katsika/pylint-issue.git && cd pylint-issue`
2. `pylint --verbose --reports=no testapp`
3. Observe results differ between platforms.

### Current behavior
When 2. command is executed on two different platforms it produces different results. Namely it does not report on `cyclic-import` on MacOs.

#### Reported errors on:

* `Linux <hostname> 4.15.0-70-generic #79-Ubuntu SMP Tue Nov 12 10:36:11 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

<pre>
$ pylint --verbose --reports=no testapp
No config file found, using default configuration
************* Module testapp.models.g
testapp/models/g.py:1:0: C0114: Missing module docstring (missing-module-docstring)
testapp/models/g.py:1:0: C0103: Class name "G" doesn't conform to PascalCase naming style (invalid-name)
testapp/models/g.py:1:0: C0115: Missing class docstring (missing-class-docstring)
testapp/models/g.py:1:0: R0903: Too few public methods (0/2) (too-few-public-methods)
************* Module testapp.models
testapp/models/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module testapp.models.t
testapp/models/t.py:1:0: C0114: Missing module docstring (missing-module-docstring)
testapp/models/t.py:4:0: C0103: Class name "T" doesn't conform to PascalCase naming style (invalid-name)
testapp/models/t.py:4:0: C0115: Missing class docstring (missing-class-docstring)
testapp/models/t.py:4:0: R0903: Too few public methods (0/2) (too-few-public-methods)
testapp/models/t.py:1:0: W0611: Unused G imported from testapp.models (unused-import)
<b>testapp/models/t.py:1:0: R0401: Cyclic import (testapp.models -> testapp.models.t) (cyclic-import)</b>

--------------------------------------------------------------------
Your code has been rated at -3.75/10 (previous run: -3.75/10, +0.00)
</pre>

* `Darwin <hostname> 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64 i386 MacBookPro15,2 Darwin`

```
$ pylint --verbose --reports=no testapp
No config file found, using default configuration
************* Module testapp.models
testapp/models/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module testapp.models.g
testapp/models/g.py:1:0: C0114: Missing module docstring (missing-module-docstring)
testapp/models/g.py:1:0: C0103: Class name "G" doesn't conform to PascalCase naming style (invalid-name)
testapp/models/g.py:1:0: C0115: Missing class docstring (missing-class-docstring)
testapp/models/g.py:1:0: R0903: Too few public methods (0/2) (too-few-public-methods)
************* Module testapp.models.t
testapp/models/t.py:1:0: C0114: Missing module docstring (missing-module-docstring)
testapp/models/t.py:4:0: C0103: Class name "T" doesn't conform to PascalCase naming style (invalid-name)
testapp/models/t.py:4:0: C0115: Missing class docstring (missing-class-docstring)
testapp/models/t.py:4:0: R0903: Too few public methods (0/2) (too-few-public-methods)
testapp/models/t.py:1:0: W0611: Unused G imported from testapp.models (unused-import)

--------------------------------------------------------------------
Your code has been rated at -2.50/10 (previous run: -2.50/10, +0.00)
```

### Expected behavior
Report exactly the same set of errors.

### pylint --version output

* Linux

```
pylint 2.4.4
astroid 2.3.3
Python 3.7.5 (default, Oct 31 2019, 13:57:23)
[GCC 5.4.0 20160609]
```

* MacOS
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.7 (default, Mar 10 2020, 15:43:33)
[Clang 11.0.0 (clang-1100.0.33.17)]
```

**Comment by Nadock** [comment]
I'm having a similar issue when running locally vs running in a GitHub action. Locally I get no reported errors, but in the action I get:
```
************* Module __module_name__
__module_name__:267:15: E1123: Unexpected keyword argument 'unwrap_response' in function call (unexpected-keyword-arg)
__module_name__:267:15: E1123: Unexpected keyword argument 'stream' in function call (unexpected-keyword-arg)
__module_name__:267:15: E1120: No value for argument 'params' in function call (no-value-for-parameter)
```

_(Real module name redacted because this is a work project.)_

---

Local is MacOS with `pylint --version`:
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.7 (default, Mar 10 2020, 15:43:33)
[Clang 11.0.0 (clang-1100.0.33.17)]
```

GitHub action has `pylint --version`:
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.7 (default, Jun  9 2020, 16:52:41) 
[GCC 7.5.0]
```

**Comment by nedbat** [comment]
I am also seeing this on coverage.py's pylint, using branch https://github.com/nedbat/coveragepy/tree/nedbat/pylint-bug-3489   I can provide more details if someone is looking at this bug.
